tests/versioning.go

```diff
@@ -4824,6 +4823,9 @@ func (s *VersioningIntegSuite) doAddAssignmentRule(ctx context.Context, tq strin
 	})
 	s.NoError(err)
 	s.NotNil(res)
+	if rule.GetRamp() == nil {
```

Comment:
why are we doing this? isn't this supposed to be done by server?

Reply:
because there's an s.eventually block that takes the test-created rule object returned by this function and waits for it to appear in the rules list. The s.Eventually fails because the test-created rule has a nil ramp but the server converts that to a 100% ramp, so the test thinks the two are not equal.

Another way to solve this problem in the test is to just create the rule object with 100% ramp in the first place, but this way we get to see that a rule inserted with Nil ramp is converted to a rule with 100% ramp.

Reply:
I'm fine with fixing the test either way; idk really which is better

Reply:
I made it so that we just create the rule with 100% ramp in the first place, since I think it makes the test less confusing. We test for correctness of the ramp=nil in the unit tests, so I think we have enough coverage for the ramp=nil case there.